test/k8sT/Policies.go

```diff
@@ -1122,6 +1117,170 @@ var _ = Describe("K8sPolicyTest", func() {
 			_ = kubectl.Exec(cmd)
 		})
 
+		SkipContextIf(helpers.DoesNotExistNodeWithoutCilium, "validates ingress CIDR-dependent L4 (FromCIDR + ToPorts)", func() {
+			var (
+				outsideNodeName, outsideIP string // k8s3 node (doesn't have agent running)
+
+				backendPod   v1.Pod // The pod that k8s3 node is hitting
+				backendPodIP net.IP
+
+				hostNodeName       string // Node that backendPod ends up on
+				hostIPOfBackendPod net.IP
+
+				policyVerdictAllowRegex, policyVerdictDenyRegex *regexp.Regexp
+			)
+
+			BeforeAll(func() {
+				RedeployCilium(
+					kubectl,
+					ciliumFilename,
+					mergeMap(deepCopyMap(daemonCfg), map[string]string{
```

Comment:
:thinking: nit: we do similar things elsewhere in the suite where we customize the daemon options. Not sure why only now we introduce the `mergeMap`/`deepCopyMap` logic (or whether we should be using this more comprehensively elsewhere).

Reply:
Yeah I was surprised to learn that we had no standard mechanism for this. At the time I didn't think too much about it, and wrote the above because it was easy enough. 

Now that I take a closer look at the codebase, there aren't a lot of places where we merge configuration over with new values. The majority of occurrences where we reuse the same `map[string]string` are simply to toggle a whole different suite of flags, and rely on the previous flags to revert back to their defaults. Rarely do you see an inheritance so to speak.  

There are a handful of inheritance occurrences, and most of them are in the same suite, `Policies`. There's one occurrence in `DatapathConfiguration`.

Given that, there's not a huge need for  `mergeMap`/ `deepCopyMap` to be used everywhere. However, having a helper function to deploy Cilium with the ability to merge configuration could useful. I'll introduce that in this suite, and see what people think.

Reply:
Implemented in e5f6bd923df820b21c2ad8eb301fefdb0ae9ce40